controllers/hyperparametertuningjob/hyperparametertuningjob_controller.go

```diff
@@ -344,6 +340,11 @@ func (r *HyperparameterTuningJobReconciler) handleSageMakerApiFailure(ctx reconc
 			ctx.Log.Info("SageMaker rate limit exceeded, will retry", "err", awsErr)
 			return RequeueAfterInterval(r.PollInterval, nil)
 		} else if awsErr.StatusCode() == 400 {
+
+			if allowRemoveFinalizer {
```

Comment:
Which case will have `noRequeue` but removeFinalizer is `false`.

Reply:
Every case except the buggy case. Like I said in description, this fix is not pretty. https://issues.amazon.com/issues/P31245685 to clean up.